modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/expression/MBLookup.java

```diff
@@ -33,27 +36,88 @@ public MBLookup(JSONArray json) {
      *   ["at", number, array]: ItemType
      * @return
      */
-    public Expression lookupAt(){ return null;}
+    public Expression lookupAt(){
+        // requires an instance of a "literal" array expression
+        if (json.size() == 3 && json.get(2) instanceof JSONArray){
+//            Expression e = ff.literal(json.get(2));
+            Expression e = parse.string(json,2);
+            if (e.evaluate(null, JSONArray.class) != null) {
+                JSONArray arr = e.evaluate(null, JSONArray.class);
+                Integer at = parse.string(json, 1).evaluate(null, Integer.class);
```

Comment:
I think all of the calls to `evaluate` in this class will need to be taken out, and delayed to render time. You likely need stick with unevaluated expressions in here, and defer them to an existing or custom function that does the evaluation at render time (similar to the comments Torben made on https://github.com/johngschulz/geotools/pull/3)